src/components/SemanticViewer.jsx

```diff
@@ -527,14 +550,33 @@ class SemanticViewer extends Component {
 										<Typography variant="subtitle1">Semantic Schema</Typography>
 									</div>
 									<br/>
-									<Typography variant="subtitle2"><strong>{this.state.selectedDataset}</strong></Typography>
-									{this.state.semanticJSON["dc:description"] === "" ? null :
-										<Typography variant="body1">{this.state.semanticJSON["dc:description"]}</Typography>}
+									<Grid container spacing={2} lg={8} md={8} xl={8} xs={12}>
+										<Grid item>
+											<Button
+												color="primary"
+												variant="contained"
+												className={classes.inlineButtons}
+												size="small"
+												onClick={this.downloadTemplate}
```

Comment:
Actaully we don't need to write custom download function. we can just use the URL. Try this
```
<Button href={`${config.semanticServiceType}/${this.state.selectedSemanticType["url"]}/template`} color="primary"
variant="contained"
className={classes.inlineButtons}
size="small"
onClick={this.downloadTemplate}
>
    Download Template
</Button>
```

Reply:
ahh I thought about this but I saw dataviewer and thought this might be the way to do it...

Reply:
Ahh, no, I need to set headers in the request. Which is why simple URL hit won't work

Reply:
It needs authorization